### PR TITLE
Unused extension point warning

### LINKED ISSUE #631: Report warning at the end of generation about unused extension points
The goal of this task is to make dokka report waringns about all extension points that has at least one declared extension and are never used during generation.

### PR DESCRIPTION
**Pull request by sswistun-vl**
Added warning when unused extension points are detected. Closes #631

### CODE REVIEW

File: testApi/src/main/kotlin/testApi/context/MockContext.kt [hunk 1]
```diff
@@ -40,6 +40,12 @@ class MockContext(
 
     override val platforms: Map<PlatformData, EnvironmentAndFacade>
         get() = testPlatforms ?: throw IllegalStateException("This mock context doesn't provide platforms data")
+    override val pointsUsed: MutableSet<ExtensionPoint<*>>
+        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
+    override val pointsPopulated: MutableSet<ExtensionPoint<*>>
+        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
+    override val unusedPoints: Collection<ExtensionPoint<*>>
+        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
```
Comment (Kordyjan) [inline testApi/src/main/kotlin/testApi/context/MockContext.kt]: Do not leave autogenerated methods stubs. If you are sure you do not want to provide any implementation throw explicit exception

File: core/src/main/kotlin/plugability/DokkaContext.kt [hunk 2]
```diff
@@ -22,6 +22,10 @@ interface DokkaContext {
     val logger: DokkaLogger
     val configuration: DokkaConfiguration
     val platforms: Map<PlatformData, EnvironmentAndFacade>
+    val pointsUsed: MutableSet<ExtensionPoint<*>>
+    val pointsPopulated: MutableSet<ExtensionPoint<*>>
```
Comment (Kordyjan) [inline core/src/main/kotlin/plugability/DokkaContext.kt]: Here you are stating that points used and points populated are part of context interface, so every user of every context is able to add or remove used or populated points in any moment. Only read-only collection of unused points should be part of the api. The rest is just implementation detail.

File: testApi/src/main/kotlin/testApi/context/MockContext.kt (hunk 1 above)
Comment (sswistun-vl) [reply testApi/src/main/kotlin/testApi/context/MockContext.kt]: Changed, now unusedPoints are mockable.

File: core/src/main/kotlin/plugability/DokkaContext.kt (hunk 2 above)
Comment (sswistun-vl) [reply core/src/main/kotlin/plugability/DokkaContext.kt]: Removed.

**Pull request merged**